testing/testrunner/utils.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"reflect"
+	"runtime"
+	"runtime/debug"
+	"syscall"
+)
+
+// This is a JSON type printed by the test binaries (not by EventsTrace), it's
+// used all over the place, so define it here to save space
+type TestPidInfo struct {
+	Tid  int64 `json:"tid"`
+	Tgid int64 `json:"tgid"`
+	Ppid int64 `json:"ppid"`
+	Pgid int64 `json:"pgid"`
+	Sid  int64 `json:"sid"`
+}
+
+// Definitions of types printed by EventsTrace for conversion from JSON
+type PidInfo struct {
+	Tid         int64 `json:"tid"`
+	Tgid        int64 `json:"tgid"`
+	Ppid        int64 `json:"ppid"`
+	Pgid        int64 `json:"pgid"`
+	Sid         int64 `json:"sid"`
+	StartTimeNs int64 `json:"start_time_ns"`
+}
+
+type CredInfo struct {
+	Ruid int64 `json:"ruid"`
+	Rgid int64 `json:"rgid"`
+	Euid int64 `json:"euid"`
+	Egid int64 `json:"egid"`
+	Suid int64 `json:"suid"`
+	Sgid int64 `json:"sgid"`
+}
+
+type TtyInfo struct {
+	Major int64 `json:"major"`
+	Minor int64 `json:"minor"`
+}
+
+type ProcessForkEvent struct {
+	ParentPids PidInfo `json:"parent_pids"`
+	ChildPids  PidInfo `json:"child_pids"`
+}
+
+type ProcessExecEvent struct {
+	Pids     PidInfo  `json:"pids"`
+	Creds    CredInfo `json:"creds"`
+	Ctty     TtyInfo  `json:"creds"`
+	FileName string   `json:"filename"`
+	Cwd      string   `json:"cwd"`
+	Argv     string   `json:"argv"`
+}
+
+type FileCreateEvent struct {
+	Pids PidInfo `json:"pids"`
+	Path string  `json:"path"`
+}
+
+type FileDeleteEvent struct {
+	Pids PidInfo `json:"pids"`
+	Path string  `json:"path"`
+}
+
+type FileRenameEvent struct {
+	Pids    PidInfo `json:"pids"`
+	OldPath string  `json:"old_path"`
+	NewPath string  `json:"new_path"`
+}
+
+type SetUidEvent struct {
+	Pids  PidInfo  `json:"pids"`
+	Creds CredInfo `json:"creds"`
+}
+
+type SetGidEvent struct {
+	Pids  PidInfo  `json:"pids"`
+	Creds CredInfo `json:"creds"`
+}
+
+func getJsonEventType(jsonLine string) string {
+	var jsonUnmarshaled struct {
+		EventType string `json:"event_type"`
+	}
+
+	err := json.Unmarshal([]byte(jsonLine), &jsonUnmarshaled)
+	if err != nil {
+		TestFail(err)
+	}
+
+	return jsonUnmarshaled.EventType
+}
+
+func runTestBin(binName string) []byte {
+	cmd := exec.Command(fmt.Sprintf("/%s", binName))
+
+	output, err := cmd.Output()
+	if err != nil {
+		fmt.Println(string(err.(*exec.ExitError).Stderr))
+		fmt.Println(string(output))
+		TestFail(fmt.Sprintf("Could not run test binary: %s", err))
+	}
+
+	return output
+}
+
+func AssertPidInfoEqual(tpi TestPidInfo, pi PidInfo) {
+	AssertInt64Equal(pi.Tid, tpi.Tid)
+	AssertInt64Equal(pi.Tgid, tpi.Tgid)
+	AssertInt64Equal(pi.Ppid, tpi.Ppid)
+	AssertInt64Equal(pi.Pgid, tpi.Pgid)
+	AssertInt64Equal(pi.Sid, tpi.Sid)
+}
+
+func AssertStringsEqual(a string, b string) {
```

Comment:
In Go, you can do this:
```suggestion
func AssertStringsEqual(a, b string) {
```

Reply:
🙏 done